Add initial map styling and interactions

aaronxsu opened the pull request:
## Overview

This PR:
- adds basemap styles
- reads neighborhood bounds from state and adds them to the map ranked by static categories
- creates a neighborhood ranking legend
- adds map interactions: neighborhood click and zoom, hover outline
- creates static user destination markers

Tagging @philippschmitt for styling and design care

### Checklist

- [ ] `fixup!` commits have been squashed
- [ ] `CHANGELOG.md` updated with summary of features or fixes, following
      [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) guidelines
- [ ] `README.md` updated if necessary to reflect the changes
- [ ] Run `./scripts/format` to lint, format, and fix the application source code.
- [ ] CI passes after rebase

### Demo

<img width="564" height="768" alt="Screenshot 2025-08-18 at 10 47 25 AM" src="https://github.com/user-attachments/assets/86d8152c-6d13-4dca-bec1-d88b5228c7dd" />

<img width="1420" height="971" alt="Screenshot 2025-08-18 at 10 46 59 AM" src="https://github.com/user-attachments/assets/8ad35e10-2dfe-404c-9e4d-7860cc290974" />



### Notes

1. Neighborhood popup upon hover isn't implemented in this one as we expected that it involves a customized popup, which should be in its own issue
2. Neighborhood rankings and user destinations are static as these parts of the app state are not implemented yet. I marked them as to-dos in the code when we have the state ready
3. The styles of the zoom controls are out of the box and I did not customize them to look like the design. We could deal with that later.

## Testing Instructions

 * Spin up the app and go to discover page
 * Make sure the map shows the basemaps
 * Make sure the neighborhood bounds show up on the map
 * Clicking on a neighborhood should zoom and highlight it
 * Hovering on a neighborhood should highlight it

Resolves https://github.com/azavea/echo-locator/issues/658

aaronxsu committed:
Add basemap styles

aaronxsu committed:
Add @turf/bbox dep

aaronxsu committed:
Add map layer styles and interactions:
- Neighborhood bounds layer colored by static categories
- Neighborhood ranking legend
- Neighborhood click and zoom interaction
- Neighborhood hover outline
- User destination static markers

philippschmitt committed:
Make sure window is never higher than visible viewport

This is relevant for iOS where the browser UI expands/collapses

philippschmitt committed:
wrap legend in another div to center it on mobile

philippschmitt committed:
Tweak legend styles

philippschmitt commented:
I tweaked the styles a little bit. Two things I couldn't change myself:

**The MapLibre attribution should be collapsed by default** (on mobile). It overlaps with the legend:

<img width="726" height="622" alt="CleanShot 2025-08-20 at 18 19 46@2x" src="https://github.com/user-attachments/assets/f0fd3e19-10a9-4004-805a-db6882a1aae1" />

I think this is possible to configure via `AttributionControl`, [see here](https://maplibre.org/maplibre-gl-js/docs/API/type-aliases/AttributionControlOptions/).


**The neighborhoods layer should be below other layers**. The neighborhood fill layer should be below the water layer, and neighborhood outline should be just above the water layer (assuming layers are ordered as I specified in the map style). This way neighborhood names are legible; neighborhood shapes are "clipped" by water; and the transit lines show up on top of the neighborhoods.

<img width="3680" height="2238" alt="CleanShot 2025-08-20 at 18 23 54@2x" src="https://github.com/user-attachments/assets/471d4554-46a1-4220-b665-79eccad6d6b3" />

rachelekm commented on src/frontend/src/pages/Discover/Map/Map.tsx:
```diff
@@ -0,0 +1,181 @@
+        }
+    };
+
+    return neighborhoodsRanked ? (
```
What do you think about moving this `neighborhoodsRanked` conditional to just around the `Source`? That way the map still renders In case there's any lag from S3 or while deriving data

aaronxsu replied:
Good idea. I addressed it in https://github.com/azavea/echo-locator/pull/690/commits/2a88f563550181d71ef784b1e9e2e5a52a452bee

rachelekm committed:
Refactor css to conditionally display mobile map/list mode

rachelekm reviewed (changes requested):
This is looking really great, I'm glad we're going with maplibre the zoom into neighborhoods looks really smooth. Ditto to Philipp's comments above and also noting I made two follow up issues to capture the work for map hover pop-ups.

I'm noticing a flicker when switching between the map and list view on mobile that seems more obvious now that we're adding a data layer. I think conditional css vs conditionally rendering the map/list might be the better choice after all (at least just for this case). I tried this out in a separate branch and it seemed to do the trick (commit https://github.com/azavea/echo-locator/commit/8a354914d55dc398110f17ff4b3e7d81e71418b6), let me know what you think!

aaronxsu commented:
@philippschmitt 

>The MapLibre attribution should be collapsed by default (on mobile). It overlaps with the legend:
>I think this is possible to configure via AttributionControl, [see here](https://maplibre.org/maplibre-gl-js/docs/API/type-aliases/AttributionControlOptions/).

I tried the Attribution Control approach but it's not collapsing as we wanted. 

```diff
diff --git a/src/frontend/src/pages/Discover/Map/Map.tsx b/src/frontend/src/pages/Discover/Map/Map.tsx
index 371f034..574690f 100644
--- a/src/frontend/src/pages/Discover/Map/Map.tsx
+++ b/src/frontend/src/pages/Discover/Map/Map.tsx
@@ -139,6 +139,7 @@ const Map = ({ isMobile = true }: Props) => {
                 onMouseLeave={onMouseLeave}
                 interactiveLayerIds={["neighborhoods", "neighborhoods-borders"]}
                 cursor="pointer"
+                attributionControl={{ compact: true }}
                 // @ts-ignore
                 mapStyle={baseMapStyle}
             >
```

It's unclear if it is enforcing it internally or not: https://maplibre.org/maplibre-gl-js/docs/API/type-aliases/AttributionControlOptions/

> If true, the attribution control will always collapse when moving the map. If false, force the expanded attribution control. The default is a responsive attribution that collapses when the user moves the map on maps less than 640 pixels wide. Attribution should not be collapsed if it can comfortably fit on the map. compact should only be used to modify default attribution when map size makes it impossible to fit default attribution and when the automatic compact resizing for default settings are not sufficient.

How about we move the legend to the top of the map container?

aaronxsu committed:
Map: adjust layer display order:
- neighborhood fill layer is below the water layer
- neighborhood outline is above the water layer

aaronxsu committed:
Map: update layer rendering conditions

aaronxsu commented:
>The neighborhoods layer should be below other layers. The neighborhood fill layer should be below the water layer, and neighborhood outline should be just above the water layer (assuming layers are ordered as I specified in the map style). This way neighborhood names are legible; neighborhood shapes are "clipped" by water; and the transit lines show up on top of the neighborhoods.

>Ditto to Philipp's comments above

These are good calls. I did not notice it. I fixed it in https://github.com/azavea/echo-locator/pull/690/commits/a70c6ff7b92e74383439c8c8be6c656cc3526e7c. I did not have much luck finding how to do it in the docs, but I learned about this `beforeId` property (not documented it seems like 🤷🏼 ) in [this post](https://qubika.com/blog/effectively-manage-layer-order-mapbox-gl-js/), which is a good read.

The display order should look fine now @rachelekm @philippschmitt 

Edit: By the way, I rebased this branch on top of `develop`. The easiest way to re-review it would be to delete the local branch, fetch, and check it out again.

<img width="555" height="776" alt="Screenshot 2025-08-26 at 2 20 13 PM" src="https://github.com/user-attachments/assets/6615be32-0970-4916-95ef-93549dc71947" />

aaronxsu commented:
>I'm noticing a flicker when switching between the map and list view on mobile that seems more obvious now that we're adding a data layer. I think conditional css vs conditionally rendering the map/list might be the better choice after all (at least just for this case). I tried this out in a separate branch and it seemed to do the trick (commit https://github.com/azavea/echo-locator/commit/8a354914d55dc398110f17ff4b3e7d81e71418b6), let me know what you think!

@rachelekm Good idea and it looks great to me! Let's have it in. Would you like to push to this branch or have it in a PR?

rachelekm reviewed (approved):
Nice job figuring out the map layers 👏  this is looking good!

> 
> @rachelekm Good idea and it looks great to me! Let's have it in. Would you like to push to this branch or have it in a PR?

I cherry-picked that commit from my branch and just pushed it up here since that seemed the simplest. Glad it worked out!

Approving on my end and think this is good to go once you and Philipp have a plan on the legend placement change

aaronxsu commented:
>I cherry-picked that commit from my branch and just pushed it up here since that seemed the simplest. Glad it worked out!

Thanks! I pushed up a small `fixup` commit to make CI happy.

@philippschmitt after we have a decision on the map legend, this will be ready to go!

aaronxsu committed:
Update map legend style

aaronxsu commented:
Update: we will move the legend up above the attribution on the mobile view and may come back to it later as budget allows.

aaronxsu committed:
Update changelog
